src/components/PeopleFilter/PeopleFilters.tsx

```diff
@@ -0,0 +1,128 @@
+import classNames from 'classnames';
+import { getSearchWith } from '../../utils/searchHelper';
+import { SearchLink } from '../SearchLink';
+
+interface Props {
+  sex: string
+  query: string
+  setSearchParams: (nextInit: string) => void
+  searchParams: URLSearchParams
+  centuries: string[]
+}
+
+export const PeopleFilters: React.FC<Props> = ({
+  sex,
+  query,
+  setSearchParams,
+  searchParams,
+  centuries,
+}) => {
+  const onQueryFilter = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newParams = { query: event.target.value.trimStart() || null };
+
+    const preparedQuery = getSearchWith(searchParams, newParams);
+
+    setSearchParams(preparedQuery);
+  };
```

Comment:
![image](https://user-images.githubusercontent.com/111217371/222140044-f8e2f69e-6751-47a7-bbc3-e2e95e366330.png)

Works with spaces for some reason, not sure why.